doc/manual/manual.tex

```diff
@@ -4181,12 +4181,34 @@ \subsubsection{A note on the syntax of formulas in input files}
 \texttt{csc()}, \texttt{sec()}, \texttt{pow()}, \texttt{log()}, 
 \texttt{erfc()}, \texttt{rand()}, and \texttt{rand\_seed()}. 
 For more detailed information, see 
-\url{http://www.dealii.org/developer/doxygen/deal.II/classFunctionParser.html#details}. 
-A common case is an if-statement that has the general form
-\texttt{if(condition, true-expression, false-expression)}. As an example, 
-the statement ``if $1<x<4$ then output 1, else output 0'' can be expressed as
-\texttt{if(1<x \&\& x<4, 1, 0)}. For more examples of
-the syntax understood, reference the documentation of the muparser library
+\url{http://www.dealii.org/developer/doxygen/deal.II/classFunctionParser.html#details}.
+
+A common need for function expresion is an if-else-statement, for example 
+``if $1<x<4$ then output 1, else output 0''.
+The muparser uses lazy-expression syntax \texttt{(if-condition ?\ true-expression :\ false-expression )}
+for if-else statements. This lazy-expression only evaluates the expression that meets the
+if-condition, rather than evaluating both expressions, which can be useful if one of the
+expression is not defined (e.g., has a divide by zero) when the if-condition is not met.
+Note it is also possible to use the syntax \texttt{if(condition, true-expression, false-expression)},
+but in this case both expressions are always evaluated.
+Therefore, the lazy-expression syntax is recommended.
+
+As a simple example using the lazy-expression syntax, the statement ``if $1<x<4$ then output 1, else output 0''
+can be expressed as \texttt{(1<x\ \&\&\ x<4\ ?\ 1\ :\ 0)}.  Multiple, nested if-else expressions
```

Comment:
Can you write this as ``(1<x \&\& x<4) ? (1) : (0)`` instead? Otherwise you are relying on the operator precedence rules between && and ? (and I have no idea what they are for the mu parser).